File: algorithms/C/Check If It Is a Good Array/Check If It Is a Good Array.java
class Solution {
    public boolean isGoodArray(int[] nums) {
        int gcd = nums[0];
        for(int i =1; i<nums.length; i++){
           gcd = GCD(gcd, nums[i]);
            if (gcd==1)
                return true;
        }
        return gcd ==1;
        
    }
    int GCD(int a, int b){
        if(b==0){
            return a;
        }
        else{
            return GCD(b, a%b);
        }
            
    }
}

File: scripts/algorithms/C/Check If It Is a Good Array/Check If It Is a Good Array.py
# Runtime: 527 ms (Top 22.03%) | Memory: 24.7 MB (Top 71.19%)
class Solution:
    def isGoodArray(self, nums: List[int]) -> bool:
        def gcd(a,b):
            while a:
                a, b = b%a, a
            return b
        return reduce(gcd,nums)==1